docs/build_docs.py

```diff
@@ -0,0 +1,880 @@
+"""Builds geoips and geoips plugin documentation."""
+
+import tempfile
+import importlib.util
+import warnings
+import logging
+import logging.handlers
+import argparse
+import shutil
+import os
+
+import brassy.actions.build_release_notes as brassy_build
+import brassy.utils.CLI  # noqa # because of a brassy bug; will be fixed in next vers
+from rich.logging import RichHandler
+from rich.traceback import install as install_rich_tracebacks
+from rich.logging import Console
+from rich.progress import Progress
+import rich_argparse
+import pygit2
+import jinja2
+import sphinx.cmd.build as sphinx_build_module
+from sphinx.ext.apidoc import main as sphinx_apidoc
+
+from update_release_note_index import main as generate_release_note_index
+
+
+def init_logger(use_rich):
+    """
+    Initialize and configure the logger.
+
+    Parameters
+    ----------
+    use_rich : bool
+        If True, sets up rich logging else use standard stream logging
+
+    Returns
+    -------
+    logger : logging.Logger
+        The configured logger instance
+    """
+    if use_rich:
+        install_rich_tracebacks()
+        logging_handlers = [RichHandler(rich_tracebacks=True)]
+    else:
+        logging_handlers = [logging.StreamHandler()]
+
+    logging.basicConfig(level=logging.DEBUG, datefmt="[%X]", handlers=logging_handlers)
+    logger = logging.getLogger("build_docs")
+    return logger
+
+
+def parse_args_with_argparse():
+    """
+    Parse and validate command-line arguments provided to the script.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed command-line arguments.
+    """
+    # Initialize parser with an example usage in the description
+    parser = argparse.ArgumentParser(
+        description=(
+            "Build Sphinx documentation for GeoIPS related packages.\n"
+            "Example: build_docs.py /path/to/repo geoips"
+        ),
+        formatter_class=rich_argparse.RichHelpFormatter,
+    )
+
+    # Required positional arguments
+    parser.add_argument(
+        "repo_path",
+        type=str,
+        help="Path to the repository (e.g., /path/to/geoips)."
+        "Must be an existing directory.",
+    )
```

Comment:
Is this necessary? For example, if we were to remove ``repo_path`` as a required argument and just require ``package_name`` instead, we could extrapolate where that package actually exists via ``os.path.dirname(str(importlib.resources.files(package_name)))``

Reply:
Here's an example in ipython.

```python
In [7]: os.path.dirname(str(importlib.resources.files("geoips")))
Out[7]: '/local/home/evan/geoips/geoips_packages/geoips'
```

Reply:
Removing path as a required argument would make this command easier to use. Just a thought!

Reply:
Great idea! Implemented here: https://github.com/NRLMMD-GEOIPS/geoips/pull/825